pennylane/templates/tensornetworks/mps.py

```diff
@@ -130,39 +130,50 @@ def __init__(
         do_queue=True,
         id=None,
     ):
-
-        self.ind_gates = compute_indices_MPS(wires, n_block_wires)
+        ind_gates = compute_indices_MPS(wires, n_block_wires)
         n_wires = len(wires)
-        shape = qml.math.shape(template_weights)[-4:]  # (n_params_block, n_blocks)
-        self.n_params_block = n_params_block
-        self.n_blocks = int(n_wires / (n_block_wires / 2) - 1)
-        self.block = block
+        n_blocks = int(n_wires / (n_block_wires / 2) - 1)
 
         if template_weights is None:
-            self.template_weights = np.random.rand(n_params_block, int(self.n_blocks))
+            template_weights = np.random.rand(n_params_block, int(n_blocks))
 
         else:
-
-            if shape[0] != self.n_blocks:
+            shape = qml.math.shape(template_weights)[-4:]  # (n_params_block, n_blocks)
```

Comment:
wait, was this a bug in the old version?

Reply:
does this mean we don't have a test for `template_weights=None`?

Reply:
Oh, good find. I didn't change the new mps templates yet, they came in while writing the PR. But I should do this :/ 

I have also no idea where the changes shown here come from, maybe a version of master that's missing in `op-refactor`? Will investigate

Reply:
Oh wait, I think I made those changes, because otherwise the tests for this template failed :) 

Basically, I need to put these variables into the dict of hyperparameters, else the `compute` methods don't get them!